infra/compile.py

```diff
@@ -749,10 +806,17 @@ def compile(tree, ctx, indent=0):
         out += "\n" + " " * indent + "}"
         return out
     elif isinstance(tree, ast.For):
+        try:
+            t = find_hint(tree.iter, "type")
+        except MissingHint:
+            t = "list"
+        assert t in ["dict", "list"]
```

Comment:
I'm fine with this code but if we ever do iterate over a `dict` I feel like it improves readability to write `for i in foo.keys()`